### PR TITLE
Add more slicing!!

### PR DESCRIPTION
**Pull request by jerwuqu**
`v` and `V` work as `s` and `S`, but are vertical! Wow!

.... also `#` and `@` for some very cool double slicing action!

### CODE REVIEW

File: src/js/HuesInfo.js [hunk 1]
```diff
@@ -43,8 +43,10 @@ const beatGlossary = [
     "= Fade and change image",
     "i Invert all colours",
     "I Invert & change image",
-    "s Slice",
-    "S Slice and change image"
+    "s Horizontal slice",
+    "S Horizontal slice and change image",
+    "v Vertical slice",
+    "V Horizontal slice and change image",
```
Comment (mon) [inline src/js/HuesInfo.js]: Horizontal 🤔

File: src/js/HuesCanvas.js [hunk 2]
```diff
@@ -572,21 +589,22 @@ class HuesCanvas {
...
+        this.sliceIsVertical = isVertical;
+
         this.generateSliceSegments();
         this.needsRedraw = true;
     }
-    
+
     generateSliceSegments() {
-        let even = 1.0 / this.sliceAvgSegments;
+        let even = 1.0 / (this.sliceAvgSegments * (this.sliceIsVertical ? 2 : 1));
```
Comment (mon) [inline src/js/HuesCanvas.js]: Add a comment about why this ternary is here

File: src/js/HuesCanvas.js [hunk 3]
```diff
@@ -232,25 +233,41 @@ class HuesCanvas {
...
-            this.offContext.drawImage(bitmap,
-                                      0 , bitmapoffset, // subsection x, y
-                                      bitmap.width, segmentBitmapHeight, // sub w/h
-                                      this.sliceDistances[i] * this.sliceDistance, drawOffset, // drawn section x, y
-                                      drawWidth, segmentDrawHeight); // drawn w/h
-            bitmapoffset += segmentBitmapHeight;
-            drawOffset += segmentDrawHeight;
+        if (this.sliceIsVertical) {
+            for(let i = 0; i < this.sliceCount; i++) {
+                let segment = this.sliceSegments[i];
```
Comment (mon) [inline src/js/HuesCanvas.js]: Duplicate code can be brought outside the `if`

File: src/js/HuesCanvas.js [hunk 4]
```diff
@@ -232,25 +233,41 @@ class HuesCanvas {
...
+                let segmentBitmapWidth = Math.round(segment * bitmap.width);
+                let segmentDrawWidth = Math.round(segment * drawWidth);
+                this.offContext.drawImage(bitmap,
+                                          bitmapoffset , 0, // subsection x, y
+                                          segmentBitmapWidth, bitmap.height, // sub w/h
+                                          drawOffset, this.sliceDistances[i] * this.sliceDistance, // drawn section x, y
+                                          segmentDrawWidth, drawHeight); // drawn w/h
+                bitmapoffset += segmentBitmapWidth;
```
Comment (mon) [inline src/js/HuesCanvas.js]: More duplicate code - standardise variable names between branches to accomplish

File: src/js/HuesCanvas.js [hunk 5]
```diff
@@ -58,14 +58,15 @@ class HuesCanvas {
         this.blurDistance = 0;
         this.xBlur = false;
         this.yBlur = false;
-
```
Comment (mon) [inline src/js/HuesCanvas.js]: I will request that you keep the "whitespace is indented" style. I'll consider nuking it across the entire project in a future commit.

**Comment by jerwuqu** [comment]
Ready for review

**Comment by mon** [comment]
I've modified the array accesses (poor readability) to use a simple object for x/y accesses. Let me know if you're happy and I'll merge!

**Comment by jerwuqu** [comment]
Looks good! Much more readable indeed.

**Pull request merged**